### ISSUE TITLE
Consistent style for references to external material

### ISSUE THREAD

**Issue by wafels** [sunpy/sunpy#3273]
### Description

We should have a consistent style in order to refer to external material.  For example, references to papers, either in the example gallery or in code docstrings, should be consistent.  The example below is one way of formatting a reference.

> Howard, R.F., Harvey, J.W. & Forgach, S. Sol Phys (1990) 130: 295. https://doi.org/10.1007/BF00156795 

The DOI should be given in preference to any other kind of link to a digital object.

Discuss!

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

### System Details

  - SunPy Version:
  - Astropy Version:
  - Python Version:
  - OS information:

**Comment by Cubostar** [comment]
I can work on this! How exactly did you format the citation @wafels ? Also, would this need to be done to all documentation in the entire code base?

**Comment by wafels** [comment]
The idea is that wherever we refer to a document source that has a DOI, the DOI should be used.  The text that goes along with the example above is just one way of referring to the document in text.  I don't have a strong preference as to how that is formatted.  If someone can suggest a standard, then I am fine with that.  For example, we could adopt the Astrophysical Journal reference style: https://journals.aas.org/references/

**Comment by mbobra** [comment]
I agree with adopting the AAS style guide. Maintaining an in-house style guide takes a lot of time.

**Comment by wtbarnes** [comment]
I'm also 👍 on using the AAS guide.

**Comment by Cadair** [comment]
We can also look at / adopt the astropy style guide: https://docs.astropy.org/en/latest/development/style-guide.html

**Comment by mbobra** [comment]
The AstroPy style guide is not complete (it is only for narratives) and does not have a style guideline for citations, which was @wafels' initial concern. The AstroPy style guide also relies on [AP style](https://www.apstylebook.com/) (proprietary), whereas AAS style is publicly available.

**Comment by wtbarnes** [comment]
I'm going to close this as this inconsistency has largely been resolved by #7837. If there are still lingering inconsistencies in our references, please reopen or open issues for specific problems.
